mne_bids/write.py

```diff
@@ -931,7 +931,17 @@ def write_raw_bids(raw, bids_path, bids_root, events_data=None,
     if raw.preload is not False:
         raise ValueError('The data should not be preloaded.')
 
-    bids_root = _path_to_str(bids_root)
+    if not isinstance(bids_path, BIDSPath):
+        raise RuntimeError('"bids_path" must be a BIDSPath object. Please '
+                           'instantiate using mne_bids.BIDSPath().')
+
+    # check root available
+    bids_root = bids_path.root
+    if bids_root is None:
+        raise ValueError('The root of the "bids_path" must be set. '
+                         'Please use `bids_path.update(root="<root>")` '
+                         'to set the root of the BIDS folder to read.')
+
```

Comment:
This is inconsistent with how we treat `root=None` elsewhere, where it essentially "working directory", right? But maybe it's good to enforce a value here so users don't accidentally end up with BIDS structures all over the place… WDYT?

Reply:
what do you have in mind? 

Like root='~/', or '~/mne_bids/'?